### ISSUE TITLE
CartesianProduct infinite sequences

### ISSUE THREAD

**Issue by 1998e663-c3b5-4cf6-92c3-7f1771ca5185** [sagemath/sage#3389]
<div id="comment:0"></div>

Hi!

The Cartesian Product iterator of infinite sequences doesn't enumerate the
every element in the product.

I tried:

```
for t in CartesianProduct(QQ,ZZ):

....:     print t

....:     

[0, 0]

[0, 1]

[0, -1]

[0, 2]
```

This is equivalent to nest for loops, which won't work.
You have to enumerate the set in a different way.

See 
http://en.wikipedia.org/wiki/Recursively_enumerable
http://en.wikipedia.org/wiki/Cantor_pairing_function

Best regards,
Michael

CC:  @sagetrac-sage-combinat

Component: **combinatorics**

_Issue created by migration from https://trac.sagemath.org/ticket/3389_

**Comment by 85eec1a4-3d04-4b4d-b711-d4db03337c41** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -5,6 +5,7 @@
 
 I tried:
 
+```
 for t in CartesianProduct(QQ,ZZ):
 
 ....:     print t
@@ -18,7 +19,7 @@
 [0, -1]
 
 [0, 2]
-
+```
 
 This is equivalent to nest for loops, which won't work.
 You have to enumerate the set in a different way.
``````

**Comment by mkoeppe** [comment]
This was fixed by #34374
